### PR TITLE
Replaced .count and .length in comment model with .size

### LINKED ISSUE #8208: Replace .count and .length in comment model with .size
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file  https://github.com/publiclab/plots2/blob/main/app/models/comment.rb , replace all instances of `.count` and `.length` with `.size`

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
$DIFF
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by Jolios**
Fixes #8208  

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -30,14 +30,14 @@ def self.comment_weekly_tallies(span = 52, time = Time.current)
...
-        .count
+        .size
     end
     weeks
   end
 
   def self.contribution_graph_making(start = Time.now - 1.year, fin = Time.now)
     date_hash = {}
-    week = start.to_date.step(fin.to_date, 7).count
+    week = start.to_date.step(fin.to_date, 7).size
```
Comment (cesswairimu) [inline app/models/comment.rb:40]: Hi @Jolios , thanks for opening this :tada: ,,,lets maintain this as `.count` `.size` will return nil here. Thanks

Comment (Jolios) [reply app/models/comment.rb]: @cesswairimu, I committed the change, could you please review it again. Thanks

File: app/models/comment.rb [hunk 2]
```diff
@@ -206,15 +206,15 @@ def unflag_comment
   end
 
   def liked_by(user_id)
-    likes.where(user_id: user_id).count > 0
+    likes.where(user_id: user_id).size > 0
```
Comment (cesswairimu) [inline app/models/comment.rb:209]: @Jolios lets refactor this further so as to remove the codeclimate warning...lets change this to `likes.where(user_id: user_id).present?` 
Then we should be good..Thanks

Comment (Jolios) [reply app/models/comment.rb]: Hi @cesswairimu , I did as you asked and the codeclimate warning is gone  but the Travs CI build fails. It seems `@related_tags` is not defined in tags controller

Comment (cesswairimu) [reply app/models/comment.rb]: No worries let me try restarting travis since its unrelated..Thanks for the fix :100:

Comment (cesswairimu) [reply app/models/comment.rb]: The issue is solved now...I will try to restart travis

**Comment by cesswairimu** [comment]
Thanks again @Jolios and welcome to PublicLab :balloon:

**Pull request merged**